server/startmonitor

```diff
@@ -1,29 +1,14 @@
 #!/bin/bash
 # Script attempts to enable a network interface into monitor mode.
 # Note: Must be run with super user privileges
-
-
-set -e 
-
-function cleanup()
-{
-    if [[ $? -ne 0 ]]; then
-        echo -e "Usage: sudo ./startmonitor.sh [interface:mon0] [channel:3] [txpower:20]"
-    else
-        echo -e "Enabled $dev in monitor mode"
-    fi
-}
-
-# keep track of the last executed command
-trap 'last_command=$current_command; current_command=$BASH_COMMAND' DEBUG
-# echo message before exiting
-trap cleanup EXIT
-
+# does accept params for monitor and channel, but defaults to mon0 and 3
 dev=${1:-mon0}
-
-ip link set $dev down
-iw dev $dev set monitor fcsfail otherbss 
-ip link set $dev up
-iw dev $dev set channel ${2:-3}
-iw dev $dev info
+chan=${2:-3}
+echo dev: $dev
+echo chan: $chan
+ip link set $dev down && \
```

Comment:
Continuing only on success is a good idea so thank you. It would be more useful if we saved the original state and reverted to it on failure though.

Reply:
hmm I'll have to think about that

Reply:
I should've looked up what `set -e` did. Apparently, it causes the script to exit immediately after any error (with some exceptions that don't apply here). So the `&&`s would have been unnecessary.

It still wouldn't have restored the original state on failure like we talked about.

Also the `trap...DEBUG` line causes that to be echoed before every command, which I suppose was a semi-hacky way to make it clear which command (if any) failed.